src/integrationTest/sam.test.ts

```diff
@@ -0,0 +1,192 @@
+}
+
+async function onDebugChanged(e: vscode.DebugSession | undefined) {
+    if (!e) {
+        return
+    }
+    assert.strictEqual(e.configuration.name, 'SamLocalDebug')
+    assert.strictEqual(e.configuration.type, debuggerType)
+    // wait for it to actually start (which we do not get an event for). 800 is
+    // short enough to finish before the next test is run and long enough to
+    // actually act after it pauses
+    await sleep(800)
+    await vscode.commands.executeCommand('workbench.action.debug.continue')
+}
+
+// Iterate through and test all runtimes
+for (const runtime of runtimes) {
+    describe(`SAM Integration tests ${runtime.name}`, async () => {
+        before(async function() {
+            // tslint:disable-next-line: no-invalid-this
+            this.timeout(TIMEOUT)
+            projectSDK = runtime.name
+            projectPath = runtime.path
+            debuggerType = runtime.debuggerType
+            // set up debug config
+            vscode.debug.onDidChangeActiveDebugSession(onDebugChanged)
+            await activateExtension('amazonwebservices.aws-toolkit-vscode')
+            console.log(`Using SDK ${projectSDK} with project in path ${projectPath}`)
+            tryRemoveProjectFolder()
+            mkdirpSync(projectFolder)
+            // this is really test 1, but since it has to run before everything it's in the before section
+            const initArguments: SamCliInitArgs = {
+                name: 'testProject',
+                location: projectFolder,
+                runtime: projectSDK as SamLambdaRuntime
+            }
+            const samCliContext = getSamCliContext()
+            await runSamCliInit(initArguments, samCliContext.invoker)
+            // Activate the relevent extensions if needed
+            if (projectSDK.includes('dotnet')) {
+                await activateExtension('ms-vscode.csharp')
+            }
+            if (projectSDK.includes('python')) {
+                await activateExtension('ms-python.python')
+            }
+            documentUri = await openSamProject()
+        })
+
+        after(async () => {
+            tryRemoveProjectFolder()
+        })
+
+        it('Generates a template with a proper runtime', async () => {
+            const fileContents = readFileSync(`${projectFolder}/testProject/template.yaml`).toString()
+            assert.ok(fileContents.includes(`Runtime: ${projectSDK}`))
+        })
+
+        it('Fails to create template when it already exists', async () => {
+            const initArguments: SamCliInitArgs = {
+                name: 'testProject',
+                location: projectFolder,
+                runtime: projectSDK as SamLambdaRuntime
+            }
+            console.log(initArguments.location)
+            const samCliContext = getSamCliContext()
+            await runSamCliInit(initArguments, samCliContext.invoker).catch((e: Error) => {
```

Comment:
If runSamCliInit succeeds, the test will pass. Is it possible to use `assertThrowsError` (from assertUtils.ts)? (I'm not sure if the error you're catching is explicitly instantiated or if it will fail the "instanceof Error" test)

If so, assertThrowsError returns the error, which you can then assert on the message.

Reply:
hm mocha really makes this annoying, in jest you can do ```expect.assertions```, but rearranged